src/admin/admin.component.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Grid,
+  Typography,
+} from '@mui/material';
+import Breadcrumbs from '../view/breadcrumbs.component';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import React from 'react';
+import Units from './units.component';
+import { BreadcrumbsInfo } from '../app.types';
+import UsageStatuses from './usageStatuses.component';
+
+export const useNavigateToAdminFunction = () => {
+  const navigate = useNavigate();
+
+  return React.useCallback(
+    (newPath: string | null) => {
+      navigate(`/admin-ims${newPath ? `/${newPath}` : ''}`);
+    },
+    [navigate]
+  );
+};
+
+// returns the admin function from the path (null when just on adminPage)
+export const useGetAdminPageName = (): string | null => {
+  const location = useLocation();
+
+  return React.useMemo(() => {
+    let adminPageName: string | null = location.pathname.replace(
+      '/admin-ims',
+      ''
+    );
+    adminPageName =
+      adminPageName === '' ? null : adminPageName.replace('/', '');
+    return adminPageName;
+  }, [location.pathname]);
+};
+
+function AdminPage() {
+  const navigateToAdminFunction = useNavigateToAdminFunction();
+  const adminPageName = useGetAdminPageName();
+
+  const adminBreadCrumbsNames: { [key: string]: [string, string] } = {
```

Comment:
```suggestion
  const adminBreadCrumbsTrails: { [key: string]: [string, string] } = {
```

Can this also be defined outside of the component function please